src/nunit.analyzers/ParallelizableUsage/ParallelizableUsageAnalyzer.cs

```diff
@@ -44,13 +44,30 @@ public override void Initialize(AnalysisContext context)
         {
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
             context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(AnalyzeCompilationStart);
             context.RegisterCompilationAction(AnalyzeCompilation);
-            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var parallelizableAttribute = context.Compilation.GetTypeByMetadataName(NUnitFrameworkConstants.FullNameOfTypeParallelizableAttribute);
+            if (parallelizableAttribute is null)
+            {
+                return;
+            }
+
+            context.RegisterSymbolAction(symbolContext => AnalyzeMethod(symbolContext, parallelizableAttribute), SymbolKind.Method);
         }
 
         private static void AnalyzeCompilation(CompilationAnalysisContext context)
         {
-            if (!TryGetAttributeEnumValue(context.Compilation, context.Compilation.Assembly,
+            var parallelizableAttribute = context.Compilation.GetTypeByMetadataName(NUnitFrameworkConstants.FullNameOfTypeParallelizableAttribute);
```

Comment:
Could `AnalyzeCompilation` not also be registered in the `RegisterCompilationStart` to remove this lookup?

Reply:
It cannot, that's why I didn't do it.